src/InvariantTheory/invariant_rings.jl

```diff
@@ -0,0 +1,185 @@
+
+mutable struct InvRing{S, T, U, V, W, X}
```

Comment:
I would prefer that as well. @thofma, you added those, anything against this?

Reply:
Be my guest. I would have thought that the field names are self-explanatory, but we can make it more redundant.

While you are at it, you might want to change `T` and `U` so that the group type is the second parameter. This will make it a bit easier when adding methods for specific group types (i.e. `InvRing{_, <: MatrixGroup}` vs `InvRing(_, _, <: MatrixGroup)`.)